ordersettler/detector.go

```diff
@@ -51,12 +48,19 @@ func DetectPendingSettlements(
 		}
 
 		for _, fill := range fills {
-			if ordersSeen != nil && ordersSeen[fill.OrderID] {
+			if ordersSeen != nil && (*ordersSeen)[fill.OrderID] {
 				continue
 			}
 
 			sourceChainID, err := config.GetConfigReader(ctx).GetChainIDByHyperlaneDomain(strconv.Itoa(int(fill.SourceDomain)))
 			if err != nil {
+				lmt.Logger(ctx).Warn(
+					"failed to get source chain ID by hyperlane domain. skipping order settlement. it may be unsettled.",
+					zap.Uint32("hyperlaneDomain", fill.SourceDomain),
+					zap.String("orderID", fill.OrderID),
+					zap.Error(err),
+				)
+				(*ordersSeen)[fill.OrderID] = true
```

Comment:
why did we make the map a pointer? a map can already be nil. you also have the potential to dereference a nil pointer here.

Reply:
sorry i didnt realize i pushed this change, this is obviously wrong

the refactor i ultimately wanted to do was for the detector to return a newOrdersSeen map, that the ordersettler can then use to update its own internal state. but not super necessary right now